exercises/react/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+#[allow(unused_variables)]
+
+// Because these are passed without & to some functions,
+// it will probably be necessary for these two types to be Copy.
+pub type CellID = ();
+pub type CallbackID = ();
+
+pub struct Reactor<T> {
+    // Just so that the compiler doesn't complain about an unused type parameter.
+    // You probably want to delete this field.
+    dummy: T,
+}
+
+// You are guaranteed that Reactor will only be tested against types that are Copy + PartialEq.
```

Comment:
open question: Can the requirement of Copy be removed? It was very difficult for me to see how.

Thinking in terms of what would have to change in the example impl. I don't have rustc to help me verify any of this right now though. I'll verify in a few hours.

If T is not `Copy` and `create_input` is made to pass an `&T`:

* then the best `Cell` can do is to also store the `&T`
* But then, if the compute functions return `T`, how will that be dealt with? Perhaps https://doc.rust-lang.org/std/borrow/enum.Cow.html ? But that means `T` has to be `ToOwned`. Still, maybe that's better
* On the other hand if the compute functions return `&T`... but how is that possible? Where can they store a `T` that will live long enough? I don't think that's possible.

If T is not `Copy` and `create_input` still passes a `T`:

* I think the cell can take ownership for `value`, and a reference for `last_value`.
* I don't understand how that can work - when `value` gets replaced, what keeps `last_value` alive?
* I guess I'll see what happens when I try it.
* Callbacks and compute functions have to take &T, that should be no problem.